### PR TITLE
Addition of pytest case coverage of `backend` and `AnalysisBase.run()` using different `n_workers` values

### LINKED ISSUE #4649: Error when `n_workers` specified twice should be tested
If both backend instance and `AnalysisBase.run(...)` call specify `n_workers`, an issue is raised. However, it's not tested, though should be.

---

              Any reason why this isn't covered by tests? If not, then at the very least please raise an issue for it to be done later.

_Originally posted by @IAlibay in https://github.com/MDAnalysis/mdanalysis/pull/4162#discussion_r1702877978_

### PR DESCRIPTION
**Pull request by talagayev**
Fixes #4649 

Adds the coverage of the `ValueError` case, where `backend` and `AnalysisBase.run()` have
different values for `n_workers`

Changes made in this Pull Request:
 - Added of `test_n_workers_conflict_raises_value_error` in `test_base.py` to cover cases, where
 both `backend` and `AnalysisBase.run()` have different values of `n_workers`

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4768.org.readthedocs.build/en/4768/

### CODE REVIEW

**Review by hmacdope — APPROVED** [review]
LGTM thanks @talagayev

File: testsuite/MDAnalysisTests/analysis/test_base.py
Comment (tylerjereddy) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: I think you can just delete this line and/or use `analysis` object below instead of repeating the `FrameAnalysis` business.

Comment (talagayev) [reply testsuite/MDAnalysisTests/analysis/test_base.py]: Ah yes my bad, I didn't notice that I still had it there, done. thanks for looking at the PR :)

**Comment by talagayev** [comment]
How is it now the case, would this and the other PR, which also only adresses a pytest be currently frozen until the release of 2.8.0 and merged after the release of 2.8.0?

**Comment by talagayev** [comment]
> LGTM thanks @talagayev

Happy to help :)

**Comment by hmacdope** [comment]
> How is it now the case, would this and the other PR, which also only adresses a pytest be currently frozen until the release of 2.8.0 and merged after the release of 2.8.0?

I believe we are in a code freeze for now, can be merged if we like but I will leave at discretion of release manager, @IAlibay.

**Comment by IAlibay** [comment]
If it's not on the 2.8 milestone, I would prefer it if we didn't merge it until after the release.

File: package/CHANGELOG
Comment (IAlibay) [inline package/CHANGELOG]: May I ask that we not add changelog entries for singular test additions. The changelog is really meant to be user facing "something has changed in the code". Large test changes are ok, but single test changes will clog up the changelog.

**Review by IAlibay — DISMISSED** [review]
Same as the other PR.

File: package/CHANGELOG
Comment (talagayev) [reply package/CHANGELOG]: Yup, adjusted it and removed the line from the Changelog :)

**Comment by talagayev** [comment]
> If it's not on the 2.8 milestone, I would prefer it if we didn't merge it until after the release.

Yup sounds good :)

**Review by RMeli — APPROVED** [review]

**Comment by talagayev** [comment]
Could it possibly connected that I asked for a re-review that @RMeli you cannot ask for a re-review?

**Pull request merged**
